components/global/mixer-adapter/wso2spadapter.go

```diff
@@ -0,0 +1,190 @@
+// nolint:lll
+// Generates the mygrpcadapter adapter's resource yaml. It contains the adapter's configuration, name,
+// supported template names (metric in this case), and whether it is session or no-session based.
+//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -a mixer/adapter/wso2spadapter/config/config.proto -x "-s=false -n wso2spadapter -t metric"
+
+package wso2spadapter
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"bytes"
+	"os"
+
+	"github.com/wso2-cellery/mesh-observability/components/global/mixer-adapter/config"
+	"istio.io/api/mixer/adapter/model/v1beta1"
+	policy "istio.io/api/policy/v1beta1"
+	"istio.io/istio/mixer/template/metric"
+	"istio.io/pkg/log"
+)
+
+type (
+	// Server is basic server interface
+	Server interface {
+		Addr() string
+		Close() error
+		Run(shutdown chan error)
+	}
+
+	// Wso2SpAdapter supports metric template.
+	Wso2SpAdapter struct {
+		listener net.Listener
+		server   *grpc.Server
+	}
+)
+
+// HandleMetric records metric entries
+func (s *Wso2SpAdapter) HandleMetric(ctx context.Context, r *metric.HandleMetricRequest) (*v1beta1.ReportResult, error) {
+
+	log.Infof("received request %v\n", *r)
+
+	var buffer bytes.Buffer
+	cfg := &config.Params{}
+
+	buffer.WriteString(fmt.Sprintf("HandleMetric invoked with:\n  Adapter config: %s\n  Instances: %s\n",
+		cfg.String(), instances(r.Instances)))
+
+	log.Infof(fmt.Sprintf("Instances: %s\n", instances(r.Instances)))
+
+	if cfg.FilePath == "" {
+		fmt.Println(buffer.String())
+	} else {
+		_, err := os.OpenFile("out.txt", os.O_RDONLY|os.O_CREATE, 0666)
+		if err != nil {
+			log.Errorf("error creating file: %v", err)
```

Comment:
Can't we return this error
fmt.Errorf("msg: %v", err)
Same applies to the other lines as well